samples/testapp/src/commonMain/kotlin/com/android/identity/testapp/ui/AppUpdateCard.kt

```diff
@@ -0,0 +1,67 @@
+package com.android.identity.testapp.ui
+
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.LinkAnnotation
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextLinkStyles
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.withLink
+import io.ktor.client.HttpClient
+import io.ktor.client.request.get
+import io.ktor.client.statement.readBytes
+import io.ktor.http.HttpStatusCode
+import org.multipaz.compose.cards.InfoCard
+import org.multipaz.testapp.BuildConfig
+import org.multipaz.testapp.platformHttpClientEngineFactory
+import org.multipaz.util.Logger
+
+private const val TAG = "AppUpdateCard"
+
+@Composable
+fun AppUpdateCard() {
+    if (BuildConfig.TEST_APP_UPDATE_URL.isEmpty()) {
+        return
+    }
+
+    val latestVersion = remember { mutableStateOf<String?>(null) }
+
+    LaunchedEffect(true) {
+        val httpClient = HttpClient(platformHttpClientEngineFactory().create())
+        val response = httpClient.get(BuildConfig.TEST_APP_UPDATE_URL)
+        if (response.status == HttpStatusCode.OK) {
+            latestVersion.value = response.readBytes().decodeToString().trim()
+            Logger.i(TAG, "Latest available version from ${BuildConfig.TEST_APP_UPDATE_WEBSITE_URL} is ${latestVersion.value}")
+        }
+    }
+
+    latestVersion.value?.compareTo(BuildConfig.VERSION)?.let {
+        if (it > 0) {
```

Comment:
We're assuming lexicographical order... If we release 1.8, 1.9, then release 1.10, this will treat 1.8 and 1.9 as more recent than 1.10. Since we're using Semantic Versioning, we may want to split on '.' and convert each value to a number, for the comparison.

Reply:
That's a good point. I've modified the PR to have testapp use https://github.com/z4kn4fein/kotlin-semver for this... also realized our current versioning scheme isn't entirely compatible with Semantic Versioning 2.0 so I also fixed that so our versions are now form `0.91.0-42.g12345678.dirty` instead of `0.91.0-42-g12345678-dirty`.